std_execution.bs

```diff
@@ -3043,6 +3042,8 @@ are all well-formed.
 
             3. If `execution::set_done(r)` has been called, calls `execution::set_done(out_r)`.
 
+            The lifetime of `op_state` ends when both `s2` and `op_state2` are destroyed.
```

Comment:
You typically do not need to keep a sender around after it has been connected to a receiver. You only need to keep the op state at that point. The op state should maintain all the state that is needed for the operation to complete.

Reply:
Yes, but in this case the shared state is also owned by the sender.

This isn't saying "you need to keep both to keep it alive"; it is saying (or at least trying to say) "either of these will keep it alive.

Do you have an idea how to reword this to make it clearer?

Reply:
Actually, this is still insufficient...

Reply:
@ericniebler please see the latest update around this line, I think this is what's actually needed.